combat-trainer.lic

```diff
@@ -1910,14 +1919,38 @@ class GameState
     @use_weak_attacks = false
   end
 
+  def use_analyze_flame?
+    @use_analyze_flame
+  end
+
+  def stop_analyze_flame
+    @use_analyze_flame = false
+  end
+
   def melee_attack_verb
```

Comment:
So, Offhand weapons don't work with this, never thought to specifically test this, but Analyze based offhand weapons as if they were brawling. I fixed this using the Brawling? check that already exists. 

The big issue is found thanks to your question. Basically, if you have a melee weapon in your hand and a set of brawling attack commands, it will still work and complete the set. If you don't, it will continue to try and chop, etc, with your fists and tell you a weapon would be a better choice. 

If someone calls combat-trainer manually, they can see this problem often, especially if they ended a combo that was melee weapon and went to brawling. It will eventually fix itself after about 90 seconds (I timed it). Analyze combos seem to be on a 90 second completion timer. 

Some ways to handle this would be to set a timer if the "Wouldn't it be better if you used a melee weapon?" was received. Set it for 90s and during that time and use the default setting. We can also check that if the analyze combo array isn't empty, that a weapon cannot be swapped. You could run into errors with this because if combat-trainer is started and gets a fresh array, the game will swap weapons to the next weapon to train, and if it happens to be brawling and the 90s timer hasn't reset, you will analyze again and get the melee weapon combo. I say, 90s of "attacking", isn't a bad thing to avoid the other headache.

Any other ideas?